tmol/io/chain_deduction.py

```diff
@@ -0,0 +1,139 @@
+import numpy
+import torch
+import scipy
+from tmol.pose.pose_stack import PoseStack
+from tmol.pose.packed_block_types import PackedBlockTypes
+from tmol.types.functional import validate_args
+from tmol.types.array import NDArray
+from tmol.utility.ndarray.common_operations import exclusive_cumsum1d
+
+
+@validate_args
+def chain_inds_for_pose_stack(
+    pose_stack: PoseStack,
```

Comment:
This is inferring chain id (for writing) from connected components in a pose stack?  You had mentioned this might be made user overridable at some point?

Reply:
Yes, the main interface function for writing a PDB file for a PoseStack now allows the user to specify their own chain indices